core/string/ustring.cpp

```diff
@@ -3470,6 +3470,10 @@ String String::replacen(const String &p_key, const String &p_with) const {
 String String::repeat(int p_count) const {
 	ERR_FAIL_COND_V_MSG(p_count < 0, "", "Parameter count should be a positive number.");
 
+	if (p_count == 1) {
```

Comment:
Not sure if this makes sense but how about
``` c++
int len = length();
if (p_count == 1 || len == 0) {
```

Reply:
Makes perfect sense, empty repeated is empty

Reply:
I'll benchmark to see if checking `is_empty()` is worth its overhead

Reply:
Since there's no resizing, the function does nothing expensive in that scenario. It does a little bit of pointless arithmetic, but it has a small impact on performance even if there are billions of repeats. So I don't think it's worth running this check every time the function is called.